### PR TITLE
Ports a PR from Shiptest. [1% Tested, should work!]

### PR DESCRIPTION
**Pull request by githubuser4141**
## About The Pull Request

Ports [Buildable blastdoors/shutters](https://github.com/shiptest-ss13/Shiptest/pull/1015) / From Shiptest!

## Why It's Good For The Game

Shutters/Blastdoors are now buildable! Yay! Adds more options to fortify or decorate your ship, which is almost always a plus.

This PR should, hopefully work from what I've played around with on local. 

## Changelog
:cl:
tweak: Blastdoors/shutters now craftable!

### CODE REVIEW

**Comment by PeriodicChaos** [comment]
I would not add the hair one as the tg people said it looks bad instead i would port their custom charter system without that hair or added with  the old hair

**Comment by githubuser4141** [comment]
pls review and/or merge :)

**Comment by HourlyLimit** [comment]
I think you should not change the hair but add to it if you could as people don't like their hair they used randomly changing

**Comment by githubuser4141** [comment]
> I think you should not change the hair but add to it if you could as people don't like their hair they used randomly changing

The sprites changed are generally superior to the ones replaced. Making them unique styles would add a ton of clutter and there's no real need for it. It's a sprite upgrade and the troublesome sprites brought up in the TG port were removed. Also, it's a LRP server where most people walk around in space suits/mechs/helmets, I doubt many people would mind.

**Comment by HourlyLimit** [comment]
> > I think you should not change the hair but add to it if you could as people don't like their hair they used randomly changing
> 
> The sprites changed are generally superior to the ones replaced. Making them unique styles would add a ton of clutter and there's no real need for it. It's a sprite upgrade and the troublesome sprites brought up in the TG port were removed. Also, it's a LRP server where most people walk around in space suits/mechs/helmets, I doubt many people would mind.

I'm sorry to say but LRP does not mean I don't care about how stuff looks also you call it a sprite upgrade I call it a replacement you should just change it so you're adding not replacing with stuff people would not want unless they could willingly pick it and yes, some people did mind it does not matter how much you remove it does not make it nicer.

**Comment by githubuser4141** [comment]
Whatever man, removes the sprite replacement for now-- just adds the blast door functionality.

File: code/game/machinery/doors/poddoor.dm [hunk 1]
```diff
@@ -15,8 +15,59 @@
...
 	damage_deflection = 70
 	poddoor = TRUE
+	assemblytype = /obj/structure/poddoor_assembly
 	var/open_sound = 'sound/machines/blastdoor.ogg'
 	var/close_sound = 'sound/machines/blastdoor.ogg'
+
+
+/obj/machinery/door/poddoor/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if((resistance_flags & INDESTRUCTIBLE) && W.tool_behaviour == TOOL_SCREWDRIVER) // This makes it so ERT members cannot cheese by opening their blast doors.
```
Comment (SuperNovaa41) [inline code/game/machinery/doors/poddoor.dm:25]: we have `screwdriver_act` `multitool_act` `crowbar_act`, etc. Use these.

File: code/game/objects/structures/poddoor_assembly.dm [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+			else
+				. +=  "<span class='notice'>The anchoring bolts are <i>unwrenched</i>, and the assembly could be <i>welded apart</i>.</span>"
+		if(AIRLOCK_ASSEMBLY_NEEDS_ELECTRONICS)
+			. += "<span class='notice'>The maintenance panel is <b>wired</b>, but the circuit slot is <i>empty</i>.</span>"
+		if(AIRLOCK_ASSEMBLY_NEEDS_SCREWDRIVER)
+			. += "<span class='notice'>The circuit is <b>connected loosely</b> to its slot.</span>"
+	. += "<span class='notice'>The assembly [welded ? "is firmly <b>welded</b> into place" : "needs to be <i>welded</i> to the floor in order to complete it"].</span>"
+	. += "<span class='notice'>There is a small placard on the assembly[doorname].</span>"
+
+/obj/structure/poddoor_assembly/attackby(obj/item/item_used, mob/user, params)
```
Comment (SuperNovaa41) [inline code/game/objects/structures/poddoor_assembly.dm:45]: same here

**Review by SuperNovaa41 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**